### PR TITLE
Create connection-schema.md

### PR DESCRIPTION
**Pull request by harshkothari410**

### CODE REVIEW

File: connection-schema.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Connection Parameter Schema in Generator Object
+```json
+{
+  "connection": {
+    "simple-auth": {
```
Comment (iamigo) [inline connection-schema.md]: is this a typo? should it be "simple-oauth"? "simple_oauth"? ...?

File: connection-schema.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+```
+## Examples
+### Argus
+```json
+{
+  "connection": {
+    "simple_oauth": {
+      "credentials": {
+        "client": {
```
Comment (iamigo) [inline connection-schema.md]: is this really meant to be an empty object? is it necessary to provide an empty object here?

File: connection-schema.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        },
+        "auth": {
+          "tokenHost": "ARGUS_HOST",
+          "tokenPath": "ARGUS_PATH"
```
Comment (iamigo) [inline connection-schema.md]: can you give an example of this argus path (but don't use our *actual* argus host name)

File: connection-schema.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+}
+```
+## Examples
+### Argus
+```json
+{
+  "connection": {
+    "simple_oauth": {
```
Comment (iamigo) [inline connection-schema.md]: is this correct? different from above?

File: connection-schema.md (hunk 2 above)
Comment (harshkothari410) [reply connection-schema.md]: Yes client is required for simple-oauth2 package.

**Pull request merged**